git-branchless/src/commands/reword.rs

```diff
@@ -267,6 +267,23 @@ fn resolve_commits_from_hashes<'repo>(
             return Ok(None);
         }
     };
+
+    // To sort the commits, we have to go from Commits -> OIDs -> CommitSet -> sort() -> Commits
+    // This helps us present a more deterministic message in the editor if, for example, the hashes
+    // specified on the CLI are in a haphazard order.
+    #[allow(clippy::needless_collect)]
+    let commit_oids = commits
+        .iter()
+        .map(|c| c.get_oid())
+        .collect::<Vec<NonZeroOid>>();
+    let commit_set = CommitSet::from_iter(
+        commit_oids
+            .into_iter()
+            .map(eden_dag::VertexName::from)
+            .map(Ok),
+    );
```

Comment:
This gave me all sorts of trouble, and I had lots of compiler issues when trying to inline the first collect. I would love to simplify it though. FWIW I just tried this:
```rust
    let commit_oids = commits
        .iter()
        .map(|c| c.get_oid())
        .collect::<Vec<NonZeroOid>>();
    let commit_set = CommitSet::from_iter(
        commit_oids.into_iter()
    );
```

And it seems like the compiler isn't connecting the dots on the `impl FromIterator<NonZeroOid> for CommitSet`, b/c it seems to be jumping directly to the version on `NameSet`, which requires a `Result<VertexName>`. 

```
error[E0271]: type mismatch resolving `<std::vec::IntoIter<NonZeroOid> as IntoIterator>::Item == Result<VertexName, DagError>`
   --> git-branchless/src/commands/reword.rs:302:22
    |
302 |     let commit_set = CommitSet::from_iter(
    |                      ^^^^^^^^^^^^^^^^^^^^ expected enum `Result`, found struct `NonZeroOid`
    |
    = note: expected enum `Result<VertexName, DagError>`
             found struct `NonZeroOid`
note: required by a bound in `NameSet::from_iter`
   --> /Users/crcarter/.cargo/registry/src/github.com-1ecc6299db9ec823/esl01-dag-0.2.1/src/nameset/mod.rs:67:25
    |
67  |         I: IntoIterator<Item = Result<VertexName>> + 'static,
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `NameSet::from_iter`

```

The code I used here is basically copied from `impl FromIterator<NonZeroOid> for CommitSet`, though, so it would be great to clean it up and simplify it. Any suggestions on how to get the `into_iter()` and `FromIterator` to connect would be great, thank you!

Reply:
I looked for other uses of `CommitSet::from_iter` in the codebase (they're all in `dag.rs`) and FWIW they all seem to be doing `.map(CommitVertex::from).map(Ok)`.

Reply:
Oops, I wrote `CommitSet::from_iter(foo.collect())` but it should be just `foo.collect()`. Anyways, you can leave this as is, and I'll update this site and the other uses in the codebase to use `.collect()` later.

Reply:
Oh, I see. I got it! `foo.into_iter().collect::<CommitSet>()` did the trick for me. I made this change here. Much nicer ... thank you!